tests/core/test_indexset.py

```diff
@@ -1,47 +1,105 @@
+import pandas as pd
+import pandas.testing as pdt
 import pytest
 
 from ..utils import all_platforms
 
 
+def df_from_list(indexsets: list):
+    return pd.DataFrame(
+        # Order is important here to avoid utils.assert_unordered_equality,
+        # which doesn't like lists
+        [
+            [
+                indexset.name,
+                indexset.elements,
+                indexset.created_at,
+                indexset.created_by,
+                indexset.run.id,
+                indexset.id,
+            ]
+            for indexset in indexsets
+        ],
+        columns=[
+            "name",
+            "elements",
+            "created_at",
+            "created_by",
+            "run__id",
+            "id",
+        ],
+    )
+
+
 @all_platforms
 class TestCoreIndexSet:
     def test_create_indexset(self, test_mp):
         run = test_mp.Run("Model", "Scenario", "new")
-        index_set_1 = run.optimization.IndexSet("IndexSet 1")
-        returned_index_set_1 = run.optimization.IndexSet("IndexSet 1")
-        assert index_set_1.id == returned_index_set_1.id
```

Comment:
Why did you remove this test case?

Reply:
Line 10 is still in there, just as line 38 now. The other two lines I removed because they are not testing the creation of IndexSets more, but getting one from the DB, which is covered in other tests. For this test to be purely about creation, I removed the `get` parts. This distinction became clear to me while writing the Scalar tests, so I went back and updated the IndexSet tests as well.